File: 0/49. Group Anagrams.py
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
      strDict = {}
      for str in strs:
        key = ''.join(sorted(str))
        if not key in strDict :
           strDict[key] = []
        strDict.get(key).append(str)

      # print(strDict.values())
      return strDict.values()

solution = Solution()
solution.groupAnagrams(['eat', 'tea', 'tan', 'ate', 'nat', 'bat'])

File: 0/49. Group Anagrams.ts
/**
 * 1A
 * 简单并查集
 * @param strs
 * @returns
 */
function groupAnagrams(strs: string[]): string[][] {
  const strMap: any = {};
  strs.forEach((str) => {
    const key = Array.from(str).sort().join('');
    if (!strMap[key]) {
      strMap[key] = [];
    }
    strMap[key].push(str);
  });
  const res: string[][] = Object.keys(strMap).map((key) => strMap[key]);
  // console.log('res', res);
  return res;
}

groupAnagrams(['eat', 'tea', 'tan', 'ate', 'nat', 'bat']);
